### DISCUSSION TITLE
JOSM and others ways to read xml and extract some infos

### DISCUSSION THREAD

**Discussion by red0bear** [mapsforge/mapsforge#1421] [General]
i was looking into https://josm.openstreetmap.de/wiki/Download since am having problems with render on desktop some boundaries don't get processed and i suspect there is some problem on pbf or osm ? or just not made yet ? boundaries levelness be see some part of maps but others no. Am testing using part of central and south America.  

Things i have been trying : 

tilesize on default;256;512;1024

Parameters.ANTI_ALIASING = true;
Parameters.FRAME_BUFFER_HA3 = true;
// Parameters.VALIDATE_COORDINATES = true;
Parameters.NUMBER_OF_THREADS = 20;
Parameters.LAYER_SCROLL_EVENT = true;
Parameters.SQUARE_FRAME_BUFFER = true;

obs 1: on 512 tilesize some sqares cant process and stay white

**Comment by devemux86** [comment]
It is not very clear what you mean or how JOSM is connected with all these unrelated parameters?

You could post a specific example that can be reproduced with [Mapsforge samples app](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-awt) with a map from [Mapsforge server](https://download.mapsforge.org/).

**Comment by red0bear** [comment]
I will post soon as possible a full example on java fx. JSOM is realted cause osmosis says ways are broken "some polygons"

**Reply by devemux86** [reply]
If you mean a custom implementation of reader or writer, we cannot examine everyone's code and apps. :slightly_smiling_face: 

Mapsforge has specific writer and reader. We all should use them in reports, so we can reproduce them.
The repository samples apps are the best examples.

**Reply by red0bear** [reply]
no no no, just a normal example using your packages; i just found way but only thing i change is internalxmlrender

**Comment by red0bear** [comment]
https://github.com/red0bear/androidmapsexample/tree/main/simplemapdesktop i did a example using javaFx and loading things i leaned looking around people post . 

i had to make some workarounds since netbeans/javafx module-info.java give a lot troubles for have same packages used by some others ones.

My issue maps don't load well and provinces borders aren't full draw ...... 
and make me think what to do next. JOSM could be more to check integrity from files o downloaded or for some troubles using osmosis/mapsforge 

you need set path to stuff on javafx modules since presets on netbeans doesnt work.
